extensions/blocks/social-previews/modal.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Social Previews modal component.
+ *
+ * Shows individual previews in modal window.
+ */
+
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button, Modal, TabPanel } from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+import { applyFilters } from '@wordpress/hooks';
+import { has } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { AVAILABLE_SERVICES } from './constants';
+
+const SocialPreviewsModal = function SocialPreviewsModal( props ) {
+	const {
+		onClose,
+		showUpgradeNudge,
+		media,
+		currentPostId,
+		title,
+		description,
+		url,
+		author,
+	} = props;
+
+	let mediaSourceUrl;
+	if ( media ) {
+		const mediaSize = applyFilters(
+			'editor.PostFeaturedImage.imageSize',
+			'post-thumbnail',
+			media.id,
+			currentPostId
+		);
+		if ( has( media, [ 'media_details', 'sizes', mediaSize ] ) ) {
+			// use mediaSize when available
+			mediaSourceUrl = media.media_details.sizes[ mediaSize ].source_url;
+		} else {
+			// get fallbackMediaSize if mediaSize is not available
+			const fallbackMediaSize = applyFilters(
+				'editor.PostFeaturedImage.imageSize',
+				'thumbnail',
+				media.id,
+				currentPostId
+			);
+			if ( has( media, [ 'media_details', 'sizes', fallbackMediaSize ] ) ) {
+				// use fallbackMediaSize when mediaSize is not available
+				mediaSourceUrl = media.media_details.sizes[ fallbackMediaSize ].source_url;
+			} else {
+				// use full image size when mediaFallbackSize and mediaSize are not available
+				mediaSourceUrl = media.source_url;
+			}
+		}
+	}
+
+	return (
+		<Modal
+			onRequestClose={ onClose }
+			title={ __( 'Social Previews', 'jetpack' ) }
+			className="jetpack-social-previews__modal"
+		>
+			{ showUpgradeNudge ? (
+				<div className="jetpack-social-previews__modal-upgrade">
+					{ /* TODO: Add upgrade nudge. */ }
```

Comment:
Intentionally left out as outside scope of this PR. We'll handle it in a followup